src/components/ListItem.jsx

```diff
@@ -2,19 +2,20 @@ import { useEffect, useState } from 'react';
 import './ListItem.css';
 import { updateItem } from '../api';
 
-export function ListItem({ name, itemId, listPath, purchaseDate }) {
-	/**TO DO: mark items as !isPurchased after 24 hours. Currently, I'm thinking
-	 * to subtract the purchaseDate from the current time at page load. Then check
-	 * if the difference is 24 hours. If less, item will still show as purchased. If
-	 * more than 24 hours difference, checkbox will be unchecked
-	 */
-	// let currentDate = new Date();
-	// console.log('check currentDate', currentDate);
-
-	// useEffect to check if item is purchased on page load so that the checkboxes are checked
+export function ListItem({
+	name,
+	itemId,
+	listPath,
+	purchaseDate,
+	nextPurchaseDate,
+}) {
 	useEffect(() => {
 		if (purchaseDate) {
-			setIsPurchased(true);
+			if (purchaseDate < nextPurchaseDate) {
+				setIsPurchased(true);
+			} else {
+				setIsPurchased(false);
+			}
```

Comment:
Could you help me understand why you compare purchasedData < nextPurchaseDate to check the 24hrs? Wouldn't purchasedDate always be less than nextPurchaseDate? 

I thought it would be comparing currentDate (when the page loads) vs nextPurchaseDate to see if it has been more than 24 hours. 

Is there something I'm not understanding correctly? Thanks!

Reply:
Ah yes, you're right. I was confusing myself with this one this morning, ha! I'll keep working on it.

Reply:
@warrenchan13 resolved! Take another look when you can, please and thank you!

Reply:
looks good!!